M2/Macaulay2/d/interp.dd

```diff
@@ -669,6 +669,22 @@ getHistory(e:Expr):Expr := (
     else WrongArgZZ());
 setupfun("getHistory", getHistory);
 
+readHistory(e:Expr):Expr := (
+    when e is filename:stringCell
+    do toExpr(readHistory(tocharstar(expandFileName(filename.v))))
+    else WrongArgString(1));
+setupfun("readHistory", readHistory);
+
+appendHistory(e:Expr):Expr := (
+    when e is s:Sequence do if length(s) != 2 then WrongNumArgs(2) else
+    when s.0 is n:ZZcell do if !isInt(n) then WrongArgSmallInteger(1) else
+    when s.1 is filename:stringCell
+    do toExpr(appendHistory(toInt(n), tocharstar(expandFileName(filename.v))))
```

Comment:
Should we raise an error if `appendHistory` has a nonzero return value?

Reply:
Maybe, but not here. This function is returning the return code intact. We _could_ check it in `Core.m2` and signal a warning, but this wasn't a priority for me.

Reply:
One advantage of checking it here is we could call `strerror` to get an more informative error message (since it returns `errno` when there's an error.)

Reply:
Does `libhistory` somehow store the meaning of the error somewhere for `strerror` to access? I didn't see anything about this in the documentation!

Reply:
Looking at the [source](https://sources.debian.org/src/readline/8.2-4/histfile.c/), I think it just passes along the value of `errno` after it's been set by one the standard library functions it calls.  So it will just be one of the various standard library error messages.

Reply:
Not sure where exactly to look in there. I'm not very familiar, but if you think it would be helpful and not depend on the system (or different versions of readline on mac, etc) feel free to make changes.

Reply:
I just pushed few commits dealing with this.  The first raises an error if `appendHistory` (or `readHistory`) fails.  That was in turn raising errors if we tried reading or appending to a nonexistent history file, so I added a couple more commits dealing with those cases.

Reply:
So for example:

```m2
i2 : readHistory "foo"
stdio:2:1:(3): error: readHistory: No such file or directory

i3 : appendHistory(0, "foo")
stdio:3:1:(3): error: appendHistory: No such file or directory
```